src/System.Numerics.Vectors/tests/Performance/Vector2/DistanceSquared.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+
+namespace System.Numerics.Tests
+{
+    public static partial class Perf_Vector2
+    {
+        [Benchmark(InnerIterationCount = VectorTests.DefaultInnerIterationsCount)]
+        public static void DistanceSquaredBenchmark()
+        {
+            const float expectedResult = 134217728.0f;
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                float result;
+
+                using (iteration.StartMeasurement())
+                {
+                    result = DistanceSquaredTest();
+                }
+
+                VectorTests.AssertEqual(expectedResult, result);
+            }
+        }
+
+        public static float DistanceSquaredTest()
+        {
+            var result = 0.0f;
+            var valueX = new Vector2(-1.0f, 1.0f);
+            var valueY = new Vector2(1.0f, -1.0f);
+
+            for (var iteration = 0; iteration < Benchmark.InnerIterationCount; iteration++)
+            {
+                valueX += VectorTests.Vector2Delta;
```

Comment:
I'll remove the adds and will leave a comment indicating that since the output is not changing, future optimizations could potentially impact the test negatively.